tests/foreman/api/test_ping.py

```diff
@@ -0,0 +1,64 @@
+"""Test Class for api ping
+
+:Requirement: Ping
+
+:CaseAutomation: Automated
+
+:CaseLevel: Component
+
+:CaseComponent: API
+
+:Assignee: gtalreja
+
+:TestType: Functional
+
+:CaseImportance: Critical
+
+:Upstream: No
+"""
+import pytest
+
+pytestmark = [pytest.mark.tier1, pytest.mark.upgrade]
+
+
+@pytest.mark.build_sanity
+def test_positive_ping(default_sat):
+    """Check if all services are running
+
+    :id: b8ecc7ba-8007-4067-bf99-21a82c833de7
+
+    :expectedresults: Overall and individual services status should be 'ok'.
+    """
+    response = default_sat.api.Ping().search_json()
+    assert response['status'] == 'ok'  # overall status
+
+    # Check that all services are OK. ['services'] is in this format:
+    #
+    # {'services': {
+    #    'candlepin': {'duration_ms': '40', 'status': 'ok'},
+    #    'candlepin_auth': {'duration_ms': '41', 'status': 'ok'},
+    #    …
+    # }, 'status': 'ok'}
+    services = response['services']
+    assert all(
+        [service['status'] == 'ok' for service in services.values()]
+    ), 'Not all services seem to be up and running!'
+
+
+def test_negative_ping_fail_status(default_sat):
+    """Negative test to verify FAIL status when ping fail
+
+    :id: 6bd2c552-29d7-4f49-81dd-f8d5ff0e7c39
+
+    :customerscenario: true
+
+    :BZ: 1835122
+
+    :expectedresults: ping fails and shows FAIL status.
+    """
+    command_out = default_sat.execute('satellite-maintain service stop --only tomcat.service')
+    assert command_out.status == 0
+    response = default_sat.api.Ping().search_json()
+    assert response['status'] == 'FAIL'
```

Comment:
should we assert more specifically on which services failed in addition to the overall status